src/benchmark.cpp

```diff
@@ -0,0 +1,151 @@
+#include <chrono>
+#include <cmath>
+#include <ctime>
+#include <iomanip>
+#include <iostream>
+#include <sstream>
+
+#include "ast/ast.h"
+#include "ast/context.h"
+#include "ast/passes/printer.h"
+#include "benchmark.h"
+
+namespace bpftrace {
+
+char TimerError::ID;
+void TimerError::log(llvm::raw_ostream &OS) const
+{
+  OS << "timer error: " << strerror(err_);
+}
+
+using time_point = std::chrono::time_point<std::chrono::steady_clock,
+                                           std::chrono::nanoseconds>;
+
+static Result<time_point> processor_time()
+{
+  struct timespec ts = {};
+  int rc = clock_gettime(CLOCK_PROCESS_CPUTIME_ID, &ts);
+  if (rc < 0) {
+    return make_error<TimerError>(errno);
+  }
+  return time_point(std::chrono::seconds(ts.tv_sec) +
+                    std::chrono::nanoseconds(ts.tv_nsec));
+}
+
+static int64_t delta(time_point start, time_point end)
+{
+  return std::chrono::duration_cast<std::chrono::nanoseconds>(end - start)
+      .count();
+}
+
+Result<> benchmark(std::ostream &out, ast::PassManager &mgr)
```

Comment:
JC, why isn't this `Result<OK>` like in the header?

Reply:
it's equivalent b/c of default template param

```
template <typename T = OK>
using Result = llvm::Expected<T>;
```